### PR TITLE
SAML/SLO: Update config & metadata

### PR DESCRIPTION
**Pull request by tung2744**
ref DEV-2059

### CODE REVIEW

File: pkg/lib/config/saml.go [hunk 1]
```diff
@@ -76,12 +104,15 @@ var _ = Schema.Add("SAMLNameIDFormat", `
 }
 `)
 
-type SAMLNameIDFormat string
-
-const (
-	SAMLNameIDFormatUnspecified  SAMLNameIDFormat = "urn:oasis:names:tc:SAML:1.1:nameid-format:unspecified"
-	SAMLNameIDFormatEmailAddress SAMLNameIDFormat = "urn:oasis:names:tc:SAML:1.1:nameid-format:emailAddress"
-)
+var _ = Schema.Add("SAMLBinding", `
```
Comment (louischan-oursky) [inline pkg/lib/config/saml.go:107]: This is unused. I guess you want to use it, right?

File: pkg/lib/config/saml.go [hunk 2]
```diff
@@ -104,29 +135,36 @@ func (p SAMLNameIDAttributePointer) MustGetJSONPointer() jsonpointer.T {
...
+	if c.NameIDFormat == samlprotocol.SAMLNameIDFormatUnspecified && c.NameIDAttributePointer == "" {
 		c.NameIDAttributePointer = "/sub"
 	}
 
 	if c.AssertionValidDuration == "" {
 		c.AssertionValidDuration = DurationString("20m")
 	}
+
+	if c.SLOBinding == "" {
+		c.SLOBinding = crewjamsaml.HTTPPostBinding
```
Comment (louischan-oursky) [inline pkg/lib/config/saml.go:166]: Use samlprotocol.SAMLBindingHTTPPost. Remove the import of `crewjamsaml` in this file.

File: pkg/lib/saml/samlprotocol/const.go [hunk 3]
```diff
@@ -16,6 +16,11 @@ var SSOSupportedBindings []SAMLBinding = []SAMLBinding{
 	SAMLBindingHTTPPost,
 }
 
+var SLOSupportedBindings []SAMLBinding = []SAMLBinding{
+	SAMLBindingHTTPRedirect,
+	SAMLBindingHTTPPost,
+}
```
Comment (louischan-oursky) [inline pkg/lib/saml/samlprotocol/const.go:22]: do we actually going to both? I am asking this because in ACS, we support HTTP-Post only.

**Review by louischan-oursky — CHANGES_REQUESTED** [review]

File: pkg/lib/saml/samlprotocol/const.go (hunk 3 above)
Comment (tung2744) [reply pkg/lib/saml/samlprotocol/const.go:22]: I think yes, as logout response is much shorter so I guess using redirect binding is possible.

File: pkg/lib/config/saml.go (hunk 1 above)
Comment (tung2744) [reply pkg/lib/config/saml.go:107]: Oh it is a mistake

File: pkg/lib/saml/samlprotocol/const.go (hunk 3 above)
Comment (tung2744) [reply pkg/lib/saml/samlprotocol/const.go:22]: Sorry this is not related to my previous comment.
This supported bindings is about what binding we accept in our slo service to receive a logout request. And yes redirect and post is accepted.
Like in SSO service, we also accept both bindings to receive a AuthnRequest.

**Comment by tung2744** [comment]
Updated, thanks!

**Review by louischan-oursky — APPROVED** [review]

**Pull request merged**